manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorker.java

```diff
@@ -0,0 +1,181 @@
+import com.redhat.service.bridge.manager.workers.Worker;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+
+public abstract class AbstractWorker<T extends ManagedResource> implements Worker<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorker.class);
+
+    @ConfigProperty(name = "event-bridge.resources.worker.max-retries")
+    int maxRetries;
+
+    @ConfigProperty(name = "event-bridge.resources.workers.timeout-seconds")
+    int timeoutSeconds;
+
+    @Inject
+    WorkManagerImpl workManager;
+
+    @Override
+    public void handleWork(Work work) {
+        //By the time VertX gets to execute an item of Work it may have been deleted.
+        if (!workManager.exists(work)) {
+            return;
+        }
+
+        T managedResource = load(work);
+        if (Objects.isNull(managedResource)) {
+            //Work has been scheduled but cannot be found. Something (horribly) wrong has happened.
+            String message = String.format("Resource of type '%s' with id '%s' no longer exists in the database.", work.getType(), work.getManagedResourceId());
+            throw new IllegalStateException(message);
+        }
+
+        boolean complete = false;
+        if (managedResource.getStatus() == ManagedResourceStatus.ACCEPTED) {
+            complete = createDependencies(managedResource).getDependencyStatus().isReady();
+        } else if (managedResource.getStatus() == ManagedResourceStatus.DEPROVISION) {
+            complete = deleteDependencies(managedResource).getDependencyStatus().isDeleted();
+        }
+
+        if (complete) {
+            workManager.complete(work);
+        } else {
+            workManager.schedule(managedResource);
+        }
+    }
+
+    @Transactional
+    protected T load(Work work) {
+        return getDao().findById(work.getManagedResourceId());
+    }
+
+    @Transactional
+    protected T setStatus(T managedResource, ManagedResourceStatus status) {
+        T resource = getDao().findById(managedResource.getId());
+        resource.setStatus(status);
+        getDao().persist(resource);
+        return resource;
+    }
+
+    @Transactional
+    protected T recordAttempt(T managedResource) {
+        T resource = getDao().findById(managedResource.getId());
+        resource.getDependencyStatus().recordAttempt();
+        getDao().persist(resource);
+        return resource;
+    }
+
+    @Transactional
+    protected T setDependencyReady(T managedResource, boolean ready) {
+        T resource = getDao().findById(managedResource.getId());
+        resource.getDependencyStatus().setReady(ready);
+        getDao().persist(resource);
+        return resource;
+    }
+
+    @Transactional
+    protected T setDependencyDeleted(T managedResource, boolean deleted) {
+        T resource = getDao().findById(managedResource.getId());
+        resource.getDependencyStatus().setDeleted(deleted);
+        getDao().persist(resource);
```

Comment:
You don't need the call to `persist` here, entity is already in the persistencecontext and will be flushed automatically by JPA/Hibernate.

Reply:
Amended. See previous comment (about use of `getDao().getEntityManager().merge(..)` for discussion.